### ISSUE TITLE
Remove TraceWriter and TextWriter support from Functions

### ISSUE THREAD

**Issue by brettsam** [Azure/azure-webjobs-sdk#1466]
We currently support ILogger, TraceWriter, and TextWriter from within functions for logging.

Now that we only use ILogger internally, it would be a good opportunity to consolidate around ILogger for functions as well. It would simplify documentation and remove some code and tests that we have to make sure they operate together nicely. In addition to removing from the code/tests, we'd need to update all sample code and templates for v3 when doing this.

**Comment by mathewc** [comment]
Agreed - I don't want to retain the two logging mechanisms at this point. I thought this was the plan all along actually.

**Comment by christianarg** [comment]
This is a bad idea: Many large codebases were written in 1x or 2x and use textwritter with this breaking change they had to be rewritten
